examples/mnist.py

```diff
@@ -1,10 +1,20 @@
 from __future__ import print_function
 
-import cPickle as pickle
 import gzip
 import itertools
-import urllib
+import pickle
 import os
+import sys
+
+PY2 = sys.version_info[0] == 2
+
+if PY2:
+    from urllib import urlretrieve
+    pickle_load = lambda f, encoding: pickle.load(f)
+else:
+    from urllib.request import urlretrieve
+    pickle_load = lambda f, encoding: pickle.load(f, encoding=encoding)
+
```

Comment:
I kinda tended towards fewer lines.  This little benchmark suggests that the bottleneck with loading isn't with cPickle versus pickle anyway:
```
>>> from timeit import timeit
>>> timeit("pickle.load(gzip.open('mnist.pkl.gz', 'rb'))", setup="import cPickle as pickle; import gzip", number=10)
5.9051759243011475
>>> timeit("pickle.load(gzip.open('mnist.pkl.gz', 'rb'))", setup="import pickle; import gzip", number=10)
5.526721000671387
```
(Don't ask me!)